### ISSUE TITLE
Sound not working in Steam

### ISSUE THREAD

**Issue by Leebre** [netblue30/firejail#3282]
Hi, I am trying to run steam in firejail on Parabola GNU/Linux (derivative of Arch). If I run it outside of firejail, it seems to work fine and XCom: Enemy Unknown starts and runs fine. However, in firejail, even if I use `--noprofile`, the sound in the Steam application doesn't work (for example, in the store videos) and the game won't launch.

**Comment by rusty-snake** [comment]
Do you use pulseaudio?

**Comment by Leebre** [comment]
@rusty-snake yes, I am.

**Comment by rusty-snake** [comment]
Related: #3165 

https://github.com/netblue30/firejail/issues/3165#issuecomment-584815612

**Comment by Leebre** [comment]
I tried `firejail --noprofile --noblacklist=/sys/module steam` and the sound still doesn't work. I am seeing the following error in the console though:

```
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
[0321/153240.860067:ERROR:alsa_util.cc(204)] PcmOpen: default,No such file or directory
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
[0321/153240.860907:ERROR:alsa_util.cc(204)] PcmOpen: plug:default,No such file or directory
```

so there is clearly some error relating to ALSA. My system is a fairly generic Dell Insipron desktop PC, x86, using a built-in sound card.

**Comment by Leebre** [comment]
I tried it with `--noblacklist=/sys/fs` as well and no joy. Something to do with ALSA must be being blocked by one of the built-in blockers?

**Comment by Leebre** [comment]
I just commented out all of the built-in blacklisted locations in `fs.c` and re-compiled. I ran `firejail --noprofile --debug-blacklists steam`, to verify that no locations were still being blacklisted - the sound still doesn't work.

**Comment by rusty-snake** [comment]
You can use `firejail --noprofile --trace=outputfile steam` to trace open, openat, fopen, access, opendir, ... or strace to trace everything. `firejail --build steam` maybe contain some hints or `firejail --noprofile --debug steam`.

**Comment by Leebre** [comment]
@rusty-snake ok, I'll give those a try and let you know what happens. Another thing I tried was running Rhythmbox in firejail (`firejail --noprofile rhythmbox`) and I got no sound with that either. So, it seems I have a general sound issue w. firejail, not just with Steam. One thing I noticed was the window title bar of Rhythmbox said it was operating as the superuser. So, I tried running it as root outside firejail and also got no sound. This might be the underlying problem - why is firejail running Rhythmbox as the superuser?

**Comment by rusty-snake** [comment]
https://github.com/netblue30/firejail/wiki/Frequently-Asked-Questions#ive-noticed-the-title-bar-in-firefox-shows-as-superuser-is-this-normal

**Comment by Leebre** [comment]
@rusty-snake thanks for the link. However, it seems strange the I get the same audio issues in firejail as I do if I try to run the application as root. I will look into how to enable sound for the root user and see if that helps with firejail.

**Comment by Leebre** [comment]
I couldn't find a good way to enable the root user to use sound with pulseaudio, so I removed pulse completely and now the sound works fine with steam in firejail (both the client and the XCom game I was trying). Again, I suspect its to do with pulse not allowing access for the root user (I'm suddenly not so much a fan of pulse any more ...)

**Comment by rusty-snake** [comment]
just remembered, have you tried `firecfg --fix-sound`?

**Comment by Leebre** [comment]
@rusty-snake yes, I ran that command after installing, per the installation instructions on firejail.wordpress.com. However, I still experienced no audio through pulseaudio with firejailed applications.

**Comment by rusty-snake** [comment]
No idea if that could change anything but you can give it a try: `firejail --noprofile --noblacklist=/sys/module --noautopulse steam`

Unfortunately I have no more ideas what the issue could be, or how to investigate it further.

**Comment by matu3ba** [comment]
Suggestion to change title to "Firejailed steam has no pulseaudio sound".

**Comment by matu3ba** [comment]
@glitsj16 Suggestion to change title to "Firejailed steam has no pulseaudio sound", because other sound apparently works.

**Comment by rusty-snake** [comment]
@matu3ba read that comment: https://github.com/netblue30/firejail/issues/3282#issuecomment-603382929

**Comment by glitsj16** [comment]
@matu3ba Why should we do that? The OP started noticing audio issues in Steam, while digging into it detected that all firejailed apps suffer from it and changed the title accordingly. Until there's a **very** good argument to do so I'm regarding the issue title the OP's prerogative.

**Comment by Leebre** [comment]
@matu3ba no, that isn't correct. I have tried with other applications that use audio and I'm not able to get sound from any of them within firejail with pulseaudio running.

@rusty-snake ok, I will give that a try when I get a chance. Although, for me, removing pulseaudio and falling back to ALSA solves the problem.

**Comment by rusty-snake** [comment]
You can try [noprofile.profie](https://gist.github.com/rusty-snake/bb234cb3e50e1e4e7429f29a7931cc72). If this does not work ... :crying_cat_face: .

**Comment by odiferousmint** [comment]
Sorry for somewhat necrobumping, but I have audio issues with Discord. It works with `--noprofile` and it works with the posted `noprofile.profile`.

I actually had audio issues with Discord inside the browser Vivaldi as well, but the following lines solved it:

```
noblacklist /sys/fs
noblacklist /sys/module
```

It does not work for the Discord app though. It works if I remove all `whitelist` lines.

For example, this one WORKS:

```
include disable-common.inc
include disable-devel.inc
include disable-interpreters.inc

noblacklist /sys/fs
noblacklist /sys/module

keep-config-pulse
```

But this one does NOT WORK:

```
include disable-common.inc
include disable-devel.inc
include disable-interpreters.inc

noblacklist /sys/fs
noblacklist /sys/module

keep-config-pulse

noblacklist ${HOME}/.config/discord
mkdir ${HOME}/.config/discord
whitelist ${HOME}/.config/discord
whitelist ${DOWNLOADS}
```

Any ideas as to why that might be the case? Keep in mind it does not work either if I comment out the `noblacklist` lines OR if the `keep-config-pulse` is missing.
